### PR TITLE
[ISEEC-972] Add "date" type to data driven componen

### PR DESCRIPTION
**Pull request by LucasGrimauth**
## Context

Jira Issue: ISEEC-972, PLT-4743

## Checklist:

- [ ] Has been verified in an integrated environment
- [ ] Has relevant unit and integration tests passing
- [ ] Has no linting, test console, or browser console errors (best effort)
- [ ] Has JSDoc comment blocks for complex code

### CODE REVIEW

**Review by Okisa — COMMENTED** [review]
Nice work Lucas!
I was testing the DateInput stories, and at the "with tooltip, label and helper text", there is no no helperText implemented at all. Maybe this is something you can add to improve the component. Overall, everything works fine!

**Review by Okisa — COMMENTED** [review]
Other thing you can add is a `dateProps` on `determineTypeProps` on `DynamicFormik` to handle any situation that we may need to modify properties for DateInputs there.

**Comment by LucasGrimauth** [comment]
> Other thing you can add is a `dateProps` on `determineTypeProps` on `DynamicFormik` to handle any situation that we may need to modify properties for DateInputs there.

Yeah, Tim and I discussed about changing the data model. I am thinking the date format pattern (e.g. "Y-m-d") for example would come from the service, not being set directly in the DynamicFormik in the client side. But I will add the `dateProps` as all other inputs have one and is important if we want to do something with the dates input it specifically.

File: src/components/DynamicFormik/DynamicFormik.stories.js
Comment (timrbula) [inline src/components/DynamicFormik/DynamicFormik.stories.js]: `min` and `max` aren't values that are currently coming from our DataDrivenComponent model in the service. I think we need to get the service team to update to include these fields. Min and max could also work for other inputs, like the number input that are currently using `minValueLength` so we should look into that if you haven't.

File: src/components/DynamicFormik/DynamicFormik.js
Comment (timrbula) [inline src/components/DynamicFormik/DynamicFormik.js:9]: I'm not seeing in this file now where its being added to the DynamicFormik validation starting on line 80 in the function `generateYupAst`. It should be added in here and we should look at the different validations we can do that make sense for a date input field

**Review by timrbula — CHANGES_REQUESTED** [review]
Looking good so far just have a few things left to consider I think.

**Review by timrbula — COMMENTED** [review]
@LucasGrimauth Before merging this in and including it in a new version of the library, we need to make sure that the data model for the services has been updated along w/ the data to use `min` instead of `minValueLength`. We should also check to make sure that any places where we are using the DataDrivenInput directly instead via DynamicFormik are updated to match (I think I did end up doing that in Flow for the Task Upgrade Modal but I am not positive).

 Other than that things are looking good! and if we can get another approval from @isakawa-carbon or @BenjaminRuby  yall can merge this in and publish a new version

**Review by timrbula — DISMISSED** [review]

**Review by Okisa — DISMISSED** [review]
All seems good and working! 🎉

**Review by BenjaminRuby — APPROVED** [review]
This looks good to me @LucasGrimauth - appreciate the effort on this and thanks for being patient with me as I get up to speed with this component!

**Pull request merged**
